libc/bionic/heap_tagging.cpp

```diff
@@ -64,19 +66,26 @@ void SetDefaultHeapTaggingLevel() {
     };
   });
 
-#if defined(USE_SCUDO)
+
   switch (heap_tagging_level) {
     case M_HEAP_TAGGING_LEVEL_TBI:
     case M_HEAP_TAGGING_LEVEL_NONE:
+#if defined(USE_SCUDO)
       scudo_malloc_disable_memory_tagging();
+#endif
+#if defined(USE_H_MALLOC)
```

Comment:
```suggestion
#elif defined(USE_H_MALLOC)
```

also, is there a check somewhere to prevent SCUDO and HMALLOC from being enabled at the same time?

Reply:
scudo and hardened_malloc are intentionally enabled at the same time on GrapheneOS.

During process init, hardened_malloc can be replaced with scudo.

See https://github.com/GrapheneOS/platform_bionic/commit/18253f960a225547515396a43c0d65660b9ebf7a